perception:
  # Basic system parameters
  processing_rate: 10.0
  visualization_rate: 15.0
  device: "auto"  # auto, cuda, cpu
  
  # Web visualization parameters
  web_interface:
    enabled: true
    port: 8080
    web_dir: "/tmp/grasp_system_live"
    update_rate: 15.0  # FPS for web updates
    image_quality: 100  # JPEG quality (1-100)
  
  # Camera parameters
  camera:
    camera1_sn: 32689769
    camera2_sn: 34783283
    transform_file_path: "/home/chris/franka_ros2_ws/src/superquadric_grasp_system/config/transform.yaml"
    resolution: HD1080  # HD720 , HD1080, HD2K
  
  # Detection parameters
  detection:
    method: "yolo"  # yolo
    yolo_model_path: "/home/chris/franka_ros2_ws/src/superquadric_grasp_system/superquadric_grasp_system/models/yolo/cubes.pt"
    confidence_threshold: 0.5
    classes: [0, 1, 2]
    class_names:
      0: "cube_1"
      1: "cube_2"
      2: "cube_3"
  
  # Point cloud parameters
  point_cloud:
    voxel_size: 0.002
    workspace_bounds: [-0.25, 0.75, -0.5, 0.5, -0.05, 2.0]
    distance_threshold: 0.4
    require_both_cameras: true
    publish_point_clouds: false

  # Pose estimation configuration
  pose_estimation:
    method: "icp"  # "icp" or "superquadric"

    # ==== SHARED PARAMETERS (used by both methods) ====
    shared:
      # Point cloud preprocessing
      poisson_reconstruction: false
      outlier_removal: true
      voxel_downsample_size: 0.002
      
      # Shared visualization (works for both ICP and superquadric)
      visualize_fused_workspace: true # useful to check if transform is correct
      enable_detected_object_clouds_visualization: true
    
      # Publishing options
      publish_poses: true
      publish_transforms: true
    
    # ==== ICP-SPECIFIC PARAMETERS ====
    icp:
      model_folder_path: "/home/chris/franka_ros2_ws/src/superquadric_grasp_system/superquadric_grasp_system/models/object_models/other"
      distance_threshold: 0.03
      max_iterations: 50
      convergence_threshold: 1e-6
      align_to_world_frame: true # Align z axis to world frame (could be useful for cubes for example)

      ## Grasp planning for ICP
      icp_grasp_planning_enabled: true
      max_grasp_candidates: 100
      
      # ICP-specific visualization
      visualize_icp_alignment: true
      visualize_grasp_poses: true

    # ==== SUPERQUADRIC-SPECIFIC PARAMETERS ====
    superquadric:
      
      # Fitting parameters
      outlier_ratio: 0.999
      use_kmeans_clustering: false # enable k-means clustering / if false: k = 1
      max_iterations: 1000
      convergence_threshold: 1e-6
      
      # Grasp planning
      sq_grasp_planning_enabled: true
      gripper_jaw_length: 0.038
      gripper_max_opening: 0.08
      
      # Superquadric-specific visualization
      enable_superquadric_fit_visualization: true
      enable_all_valid_grasps_visualization: true
      enable_best_grasps_visualization: true
      enable_support_test_visualization: true
      enable_collision_test_visualization: true


